src/Microsoft.DotNet.Build.Tasks/PackageFiles/init-tools.cmd

```diff
@@ -0,0 +1,34 @@
+@echo off
+setlocal
+
+set PROJECT_DIR=%~1
+set DOTNET_CMD=%~2
+set TOOLRUNTIME_DIR=%~3
+set TOOLS_DIR=%~dp0
+
+if not exist "%PROJECT_DIR%" (
+  echo ERROR: Cannot find project root path at [%PROJECT_DIR%]. Please pass in the source directory as the 1st parameter.
+  exit /b 1
+)
+
+if not exist "%TOOLS_DIR%" (
+  echo ERROR: Cannot find tools path at [%TOOLS_DIR%]. Please pass in the tools directory as the 2nd parameter.
+  exit /b 1
+)
+
+if not exist "%DOTNET_CMD%" (
+  echo ERROR: Cannot find dotnet command at [%DOTNET_CMD%]. Please pass in the dotnet command as the 3rd parameter.
+  exit /b 1
+)
+
+set tempDirectory=%cd%
+cd %TOOLS_DIR%\tool-runtime\
+call %DOTNET_CMD% restore
+call %DOTNET_CMD% publish -f dnxcore50 -r win7-x64 -o %TOOLRUNTIME_DIR%
+cd %tempDirectory%
+
+xcopy /E /Y /Q %TOOLS_DIR%..\lib %TOOLRUNTIME_DIR%
+
+call type NUL > %TOOLRUNTIME_DIR%\ToolRuntime.semaphore
```

Comment:
I was trying to do this now and found a problem with it. The project.json file is being located at $(ToolRuntimePath)/<buildToolsversion>/project.json. The only reason why I was still generating the semaphore file is because in our toolruntime targets, we have it in our 'EnsureBuildToolsRuntime' target. The problem is that if I were to  use project.json instead, that would mean that we would need to know the buildToolsVersion in the msbuild side. I'm not sure how to get around that. Would it be fine if I only check if $(ToolRuntimePath) exists in that target instead of checking the semaphore file?